tests/tests.go

```diff
@@ -289,6 +298,30 @@ func ChangeManagerLeadership() {
 	}, 2*time.Minute, 3*time.Second).Should(Equal(corev1.ConditionTrue), "should have a leader manager pod with ready condition")
 }
 
+func waitManagerDeploymentReady() {
+	By(fmt.Sprintf("Waitting for manager deployment to be ready"))
+	Eventually(func() (corev1.ConditionStatus, error) {
+		managerDeployment, err := testClient.KubeClient.AppsV1().Deployments(managerNamespace).Get(context.TODO(), names.MANAGER_DEPLOYMENT, metav1.GetOptions{})
+		if err != nil {
+			return corev1.ConditionUnknown, err
+		}
+		pods, err := getKubemacpoolPods()
+		if err != nil {
+			return corev1.ConditionUnknown, err
+		}
+		if int32(len(pods.Items)) != *managerDeployment.Spec.Replicas {
```

Comment:
why not just compare deployment.available to deployment.replicas? I mean like in the original implementation:
```
if managerDeployment.Status.ReadyReplicas != numOfReplica {
```

Reply:
Doing changes from 2 to 1 was not working, since it has already 1 ready but the other pod is still terminating so it continues, checking number of pods is exactly what we need and with current implementation Deployment Available is usable too.